ubidots/ubidots_in.js

```diff
@@ -1,94 +1,200 @@
-      educational: 'things.ubidots.com'
-    }
-
-    var labelDevice = n.device_label || n.label_device
-    var labelVariable = n.label_variable
-    var endpointUrl = ENDPOINTS_URLS[n.tier] || ENDPOINTS_URLS.business
-    var token = n.token
-
-    getClient(self, endpointUrl, labelDevice, labelVariable, token)
-
-    this.on('error', function () {
+      business: "industrial.api.ubidots.com",
+      educational: "things.ubidots.com"
+    };
+    var useTLS = config.tls_checkbox_in;
+    var labelDevice = config.device_label;
+    var labelVariable = config["label_variable_1"];
+    var endpointUrl = ENDPOINTS_URLS[config.tier] || ENDPOINTS_URLS.business;
+    var token = config.token;
+
+    var topics = {};
+    topics = getSubscribePaths(config);
+
+    getClient(
+      self,
+      topics,
+      useTLS,
+      endpointUrl,
+      labelDevice,
+      labelVariable,
+      token
+    );
+
+    this.on("error", function(msg) {
+      console.log("Client: Inside error function", msg);
       if (self.client !== null && self.client !== undefined) {
-        self.client.end(true, function () {})
+        self.client.end(true, function() {});
       }
-      self.status({ fill: 'red', shape: 'ring', text: 'ubidots.error_connecting' })
-    })
-
-    this.on('close', function () {
+      self.status({
+        fill: "red",
+        shape: "ring",
+        text: "ubidots.error_connecting"
+      });
+    });
+
+    this.on("close", function() {
       if (self.client !== null && self.client !== undefined) {
-        self.client.end(true, function () {})
+        self.client.end(true, function() {});
       }
-    })
+    });
 
-    this.on('input', function (msg) {
+    this.on("input", function(msg, send, done) {
       try {
-        this.send(msg)
+        send(msg);
       } catch (err) {
-        this.error(err, msg)
+        console.log("Error in client when sending data to debug node,", err);
+        this.error(err, msg);
       }
-    })
+      if (done) {
+        done();
+      }
+    });
   }
 
-  RED.nodes.registerType('ubidots_in', UbidotsNode)
+  RED.nodes.registerType("ubidots_in", UbidotsNode);
+};
+
+function getSubscribePaths(config) {
+  var paths = [];
+  var labelString = "label_variable_";
+  var completeLabelString = "";
+  var checkboxString = "checkbox_variable_";
+  var checkboxString2 = "_last_value";
+  var completeCheckboxString = "";
+  //use customtopics
+  if (config.custom_topic_checkbox) {
+    for (var i = 1; i < 11; i++) {
+      completeLabelString = labelString + i.toString();
+      if (!(config[completeLabelString] === "")) {
+        paths.push("/v1.6/devices/" + config[completeLabelString]);
+      }
+    }
+  } else {
+    for (var i = 1; i < 11; i++) {
+      completeLabelString = labelString + i.toString();
+      completeCheckboxString = checkboxString + i.toString() + checkboxString2;
+      if (!(config[completeLabelString] === "")) {
+        //last Value is true
```

Comment:
Leave the comment below, this is a bit cryptic

Reply:
el comment "last VAlue is true" is cryptic